pkg/cmd/miscellaneous.go

```diff
@@ -350,17 +361,84 @@ func getRole() string {
 }
 
 /*
-getTechnicalID returns the Technical Id, which used as clustername of the shoot cluster
+getTargetMapInfo retun garden,project,seed,shoot,TechnicalID to global getTargetInfo
 */
-func getTechnicalID() string {
+func getTargetMapInfo() map[string]string {
 	var target Target
 	ReadTarget(pathTarget, &target)
 
-	gardenClientset, err := target.GardenerClient()
-	checkError(err)
-	project, err := gardenClientset.CoreV1beta1().Projects().Get(target.Stack()[1].Name, metav1.GetOptions{})
-	checkError(err)
-	shoot, err := gardenClientset.CoreV1beta1().Shoots(*project.Spec.Namespace).Get(target.Stack()[2].Name, metav1.GetOptions{})
-	checkError(err)
-	return shoot.Status.TechnicalID
+	for _, t := range target.Stack() {
+		getTargetInfo[string(t.Kind)] = string(t.Name)
+	}
+
+	// user for case garden-[project]-seed-shoot
+	if isTargetFilsHas("seed", "shoot") {
+		delete(getTargetInfo, "seed")
+
+		shootname := getTargetInfo["shoot"]
+		project, err := getProjectObject()
+		checkError(err)
+		shoot, err := getShootObject(*project.Spec.Namespace, shootname)
+		checkError(err)
+		getTargetInfo["seed"] = *shoot.Spec.SeedName
+		getTargetInfo["shootTechnicalID"] = shoot.Status.TechnicalID
+
+		shoot, err = getShootObject("garden", getTargetInfo["seed"])
+		checkError(err)
+		getTargetInfo["seedTechnicalID"] = shoot.Status.TechnicalID
+	}
+
+	// user for case garden-project-shoot
+	if isTargetFilsHas("project", "shoot") {
+		shootname := getTargetInfo["shoot"]
+		project, err := getProjectObject()
+		checkError(err)
+		shoot, err := getShootObject(*project.Spec.Namespace, shootname)
+		checkError(err)
+		getTargetInfo["seed"] = *shoot.Spec.SeedName
+		getTargetInfo["shootTechnicalID"] = shoot.Status.TechnicalID
+
+		if getRole() == "operator" {
+			shoot, err = getShootObject("garden", getTargetInfo["seed"])
+			checkError(err)
+			getTargetInfo["seedTechnicalID"] = shoot.Status.TechnicalID
+		}
+
+	}
+
+	return getTargetInfo
+}
+
+/*
+lookup Target Kind ("kind value") return Target Name "name value" from target file ~/.garden/sessions/plantingSession/target
+*/
+func lookupTargetNameFromKind(Name string) string {
+	var target Target
+	ReadTarget(pathTarget, &target)
+	for _, t := range target.Stack() {
+		if string(t.Kind) == Name {
+			return string(t.Name)
+		}
+	}
+	return ""
+}
+
+/*
+check if target Kind is exist in target file ~/.garden/sessions/plantingSession/target
+*/
+func isTargetFilsHas(args ...string) bool {
+	var target Target
+	ReadTarget(pathTarget, &target)
+	var kindString string
+
+	for _, t := range target.Stack() {
+		kindString = string(t.Kind) + "," + kindString
+	}
+
+	for _, t := range args {
+		if !(strings.Contains(kindString, t)) {
+			return strings.Contains(kindString, t)
+		}
+	}
```

Comment:
```go
	for _, t := range target.Stack() {
		kindString = string(t.Kind) + "," + kindString
	}

	for _, t := range args {
		if !(strings.Contains(kindString, t)) {
			return strings.Contains(kindString, t)
		}
	}
```
This is also a fragile solution, even if it seems to "work" for now. 
I guess adding the kinds into a `map[string]struct{}` and then check for existence is cleaner, instead of concatenating it to one string and then check if the kind is included in the string.

Reply:
done